### ISSUE TITLE
Usage of FusedRMSNorm with T5 model

### ISSUE THREAD

**Issue by sh0tcall3r** [huggingface/transformers#25825]
Hello!
In HF T5 model description I found that if I had installed Nvidia apex package, T5 model would automatically use FusedRMSNorm instead of T5LayerNorm and hence inference would increase.
I just installed apex and using mt0-xxl model in 8bit loading mode, which I've already fine-tuned. But I see no difference neither in layers description (there are still T5LayerNorm layers in model named children list), nor in inference speed. 
I know that in case of really models, speed increase not dramatically, but am I doing everything right?

I have cuda 11.3
pytorch 1.12.1
apex 0.1

**Comment by ydshieh** [comment]
Hi @sh0tcall3r 

I assume you use https://huggingface.co/bigscience/mt0-xxl/blob/main/config.json

It is `MT5ForConditionalGeneration` whose definition is in `src/transformers/models/mt5/modeling_mt5.py`. In that file, `FusedRMSNorm` is not used.

**Comment by ydshieh** [comment]
However, I think what you see should be `MT5LayerNorm` rather than `T5LayerNorm`. Maybe it is a typo in your description ?

**Comment by sh0tcall3r** [comment]
> However, I think what you see should be `MT5LayerNorm` rather than `T5LayerNorm`. Maybe it is a typo in your description ?

Yes, indeed. I see MT5LayerNorm. And yes, you're absolutely right, it's MT5ForConditionalGeneration.
Is there any option to use apex for speed up of inference in this case?
Or is there any option of inference speed up without apex, but with some other methods/libraries/packages?

**Comment by ydshieh** [comment]
You can try to modify the modeling file of mt5 (see what has been done in t5 file).
Or you can use `load_in_8bit` or `load_in_4bit`, see `docs/source/en/main_classes/quantization.md`.

Or see if [this](https://twitter.com/younesbelkada/status/1696478075721302143) works for mt5 :-)

**Comment by sh0tcall3r** [comment]
> You can try to modify the modeling file of mt5 (see what has been done in t5 file). Or you can use `load_in_8bit` or `load_in_4bit`, see `docs/source/en/main_classes/quantization.md`.
> 
> Or see if [this](https://twitter.com/younesbelkada/status/1696478075721302143) works for mt5 :-)

Got it, will do, thanks!

**Comment by ydshieh** [comment]
Let me close this issue - I think what could be answered are all given :-)

**Comment by sh0tcall3r** [comment]
Sure, I agree
